File: src/IsbInterop.CSharp/ServiceForms/IEdmsExplorer.cs
using IsbInterop.Accessory;
using IsbInterop.DataTypes.Enumerable;
using IsbInterop.Folders;

namespace IsbInterop.ServiceForms
{
  /// <summary>
  /// Проводник системы.
  /// </summary>
  public interface IEdmsExplorer : IIsbComObjectWrapper
  {
    /// <summary>
    /// Тип отображения дерева проводника.
    /// </summary>
    TItemShow ShowTree { get; set; }

    /// <summary>
    /// Получает содержимое дерева проводника.
    /// </summary>
    /// <returns>Объект IContents, состоящий из элементов типа IFolderInfo.</returns>
    /// <remarks>
    /// Значением свойства TreeRootContents может быть:
    /// * объект IFolderInfo;
    /// * массив объектов IFolderInfo;
    /// * объект IContents, состоящий из элементов типа IFolderInfo.
    /// </remarks>
    IContents<IFolderInfo> GetTreeRootContents();

    /// <summary>
    /// Устанавливает содержимое дерева проводника.
    /// </summary>
    /// <param name="value">Значение.</param>
    void SetTreeRootContents(IFolderInfo value);

    /// <summary>
    /// Устанавливает содержимое дерева проводника.
    /// </summary>
    /// <param name="value">Значение.</param>
    void SetTreeRootContents(IContents<IFolderInfo> value);

    /// <summary>
    /// Показывает проводник.
    /// </summary>
    void Show();
  }
}

File: src/IsbInterop/ServiceForms/IEdmsExplorer.vb
Imports IsbInterop.Accessory
Imports IsbInterop.DataTypes.Enumerable
Imports IsbInterop.Folders

Namespace ServiceForms

  ''' <summary>
  ''' Проводник системы.
  ''' </summary>
  Public Interface IEdmsExplorer
    Inherits IBaseIsbObject

    ''' <summary>
    ''' Тип отображения дерева проводника.
    ''' </summary>
    Property ShowTree As TItemShow

    ''' <summary>
    ''' Получает содержимое дерева проводника.
    ''' </summary>
    ''' <returns>Объект IContents, состоящий из элементов типа IFolderInfo.</returns>
    ''' <remarks>
    ''' Значением свойства TreeRootContents может быть:
    ''' * объект IFolderInfo;
    ''' * массив объектов IFolderInfo;
    ''' * объект IContents, состоящий из элементов типа IFolderInfo.
    ''' </remarks>
    Function GetTreeRootContents() As IContents(Of IFolderInfo)

    ''' <summary>
    ''' Устанавливает содержимое дерева проводника.
    ''' </summary>
    ''' <param name="value">Значение.</param>
    Sub SetTreeRootContents(value As IFolderInfo)

    ''' <summary>
    ''' Устанавливает содержимое дерева проводника.
    ''' </summary>
    ''' <param name="value">Значение.</param>
    Sub SetTreeRootContents(value As IContents(Of IFolderInfo))

    ''' <summary>
    ''' Показывает проводник.
    ''' </summary>
    Sub Show()
  End Interface
End NameSpace